src/traversing.js

```diff
@@ -168,10 +159,18 @@ jQuery.each({
 		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
 	}
 }, function( name, fn ) {
+	var // Sorting (which implicitly de-dupes) is required for non-adjacent traversals
+		sort = name !== "prev" && name !== "next",
```

Comment:
It was based on real world usage of jQuery UI. The graphs from http://blog.jquery.com/2011/01/31/jquery-15-released/ (under Adjacent Traversal Performance) should still be relevant today, though the important data (IE performance improvement) is hard to see because of the crazy performance improvement in Chrome. The problem was that old IE (where we have to take a slow path in sorting, if I remember correctly) is noticeably slow with a few hundred elements.

Granted the specific scenario that prompted this was terrible UX (500 results in an autocomplete menu), but I do recall this impacting quite a few different scenarios that we had. However, this was for `.children()`, not `.next()` or `.prev()`. While we do have cases where we do `multipleElements.next()`, I'm not sure the numbers ever get very high. I'd have to look through all our uses of these methods to find out.